tools/@aws-cdk/prlint/lint.ts

```diff
@@ -16,28 +13,144 @@ class LinterError extends Error {
   }
 }
 
-function createGitHubClient() {
-  const token = process.env.GITHUB_TOKEN;
+interface Test {
+  test: (pr: any, files: any[]) => any;
+}
+
+class ValidateResult {
+  constructor(public errors: string[]) {
+    this.errors = errors;
+  }
 
-  if (token) {
-    console.log("Creating authenticated GitHub Client");
-  } else {
-    console.log("Creating un-authenticated GitHub Client");
+  public isValid() {
+    return this.errors.length === 0;
   }
+}
+
+interface ValidationCollectorOptions {
```

Comment:
Can you add docstrings to all the interfaces, classes, properties, etc.

Reply:
Yep. Forgot to do that. Adding.